From 49f45d5b2f905992bce391357dd81a2c1f4dad40 Mon Sep 17 00:00:00 2001
From: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
Date: Mon, 10 Feb 2025 19:18:02 -0800
Subject: [PATCH 11/28] PENDING: net: stmmac: dwmac-qcom-ethqos: Configure host
 DMA width for SA8797P

SA8797P can support DMA width of up to 40bits, update the same
value within ethernet driver so that it will give the flexibility
to map higher address range as well.

Change-Id: Ib38f1fd23bae6bd246ed63027de872cd762d3824
Signed-off-by: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
index 3457de21c2abc..c1ffb9cea4065 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
@@ -344,7 +344,7 @@ static const struct ethqos_emac_driver_data emac_v6_6_0_data = {
 	.rgmii_por = emac_v6_6_0_por,
 	.num_rgmii_por = ARRAY_SIZE(emac_v6_6_0_por),
 	.rgmii_config_loopback_en = false,
-	.dma_addr_width = 32,
+	.dma_addr_width = 40,
 	.link_clk_name = "phyaux",
 	.has_xgmac = true,
 	.has_hdma = true,
